### PR TITLE
Spaces: separate items based on the space (team)

### PR DESCRIPTION
**Pull request by demoth**
Now only relevant items (item.space == user.space) to the user is shown. 

 - todo: manage spaces (for now only default space is created)

### CODE REVIEW

File: src/main/kotlin/org/dnj/memoria/controllers/UserController.kt
Comment (demoth) [inline src/main/kotlin/org/dnj/memoria/controllers/UserController.kt:1]: leftover from a previous PR

File: src/main/kotlin/org/dnj/memoria/Initializer.kt [hunk 1]
```diff
@@ -1,24 +1,46 @@
...
+                userRepository.save(it)
+            }
+            
+            itemRepository.findAll().forEach { 
+                it.space = defaultSpace
+                itemRepository.save(it)
+            }
+        }
+        
         println("Initialized!")
```
Comment (denolia) [inline src/main/kotlin/org/dnj/memoria/Initializer.kt:35]: maybe should be logger.info too?

File: src/main/kotlin/org/dnj/memoria/model.kt [hunk 2]
```diff
@@ -6,12 +6,31 @@ import org.springframework.data.mongodb.core.mapping.DBRef
...
 import java.util.Date
 
+@Document
+data class Space(
+    val name: String,
+    val description: String? = null,
+    val created: Date = Date(),
+    @field:Id val id: String? = null,
+) {
+    fun toDto() = SpaceDto(id!!, name)
```
Comment (denolia) [inline src/main/kotlin/org/dnj/memoria/model.kt:16]: shouldn't we handle the null case?

**Review by denolia — APPROVED** [review]
:space_invader:

File: src/main/kotlin/org/dnj/memoria/model.kt (hunk 2 above)
Comment (demoth) [reply src/main/kotlin/org/dnj/memoria/model.kt:16]: should!
theoretically this dto could only be constructed from a db object. But I haven't yet come up with a proper code constraints

**Pull request merged**